spec/acceptance/facts_spec.rb

```diff
@@ -0,0 +1,50 @@
+require 'spec_helper_acceptance'
+
+describe 'sensu::backend class', unless: RSpec.configuration.sensu_cluster do
+  backend = only_host_with_role(hosts, 'sensu_backend')
+  agent = only_host_with_role(hosts, 'sensu_agent')
+  context 'backend facts' do
+    it 'should work without errors' do
+      pp = <<-EOS
+      include ::sensu::backend
+      EOS
+
+      # Run it twice and test for idempotency
+      apply_manifest_on(backend, pp, :catch_failures => true)
+      apply_manifest_on(backend, pp, :catch_changes  => true)
+      # Simulate plugin sync
+      fact_path = File.join(File.dirname(__FILE__), '../..', 'lib/facter')
+      scp_to(backend, fact_path, '/opt/puppetlabs/puppet/cache/lib/')
+    end
+
+    it "should have backend facts" do
```

Comment:
perhaps we should test that before including the class, though after "plugin sync", that the facts do not exist.

Reply:
The only way to do that would be to uninstall the various components first because tests in other files loaded first will install everything so when `facts_spec.rb` runs the backend and agent are already installed.  I only included the classes in this file to allow it to be run directly via `bundle exec rspec spec/acceptance/facts_spec.rb` which is sometimes handy to speed up testing.

Reply:
ok, makes sense to me